community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/ast/ResolvedFunctionInvocation.scala

```diff
@@ -80,7 +80,14 @@ case class ResolvedFunctionInvocation(qualifiedName: QualifiedName,
               arg.semanticCheck(SemanticContext.Results) chain arg.expectType(field.typ.covariant)
           }.foldLeft(success)(_ chain _)
         } else {
-          error(_: SemanticState, SemanticError(s"Function call does not provide the required number of arguments ($expectedNumArgs)", position))
+          val msg = (if (signature.inputSignature.isEmpty) "arguments"
+          else if (signature.inputSignature.size == 1) s"argument with type ${signature.inputSignature.head.typ.toNeoTypeString}"
+          else s"arguments of type ${signature.inputSignature.map(_.typ.toNeoTypeString).mkString(", ")}") +
+            signature.description.map(d => s"${System.lineSeparator()}Description: $d").getOrElse("")
+          error(_: SemanticState, SemanticError( s"""Function call does not provide the required number of arguments: expected $expectedNumArgs got $actualNumArgs.
+             |
+             |Function ${signature.name} has signature: $signature
+             |meaning that it expects $expectedNumArgs $msg""".stripMargin, position))
```

Comment:
It would have to be a function that would need to take a lot of arguments and when trying it out I think it just makes the code harder to read. I actually prefer the duplication.  Ping me if you have a suggestion, so far I have only been able to make it worse